src/Controller/Component/SecurityComponent.php

```diff
@@ -174,17 +180,38 @@ public function requireAuth($actions)
      *
      * @param \Cake\Controller\Controller $controller Instantiating controller
      * @param string $error Error method
+     * @param \Cake\Controller\Exception\SecurityException $exception Additional debug info describing the cause
      * @return mixed If specified, controller blackHoleCallback's response, or no return otherwise
      * @see \Cake\Controller\Component\SecurityComponent::$blackHoleCallback
      * @link http://book.cakephp.org/3.0/en/controllers/components/security.html#handling-blackhole-callbacks
      * @throws \Cake\Network\Exception\BadRequestException
      */
-    public function blackHole(Controller $controller, $error = '')
+    public function blackHole(Controller $controller, $error = '', SecurityException $exception = null)
     {
         if (!$this->_config['blackHoleCallback']) {
-            throw new BadRequestException('The request has been black-holed');
+            $this->_throwException($exception);
         }
-        return $this->_callback($controller, $this->_config['blackHoleCallback'], [$error]);
+        return $this->_callback($controller, $this->_config['blackHoleCallback'], [$error, $exception]);
+    }
+
+    /**
+     * Check debug status and throw an Exception based on the existing one
+     *
+     * @param \Cake\Controller\Exception\SecurityException $exception Additional debug info describing the cause
+     * @throws \Cake\Network\Exception\BadRequestException
+     * @return void
+     */
+    protected function _throwException($exception = null)
+    {
+        $defaultMessage = 'The request has been black-holed';
+        if ($exception !== null) {
+            if (!Configure::read('debug')) {
+                $exception->setReason($exception->getMessage());
+                $exception->setMessage($defaultMessage);
```

Comment:
What if the exception doesn't implement these methods? You should probably type check or ensure the methods exist first.